nupic/research/frameworks/pytorch/model_utils.py

```diff
@@ -36,11 +36,23 @@ def train_model(
     optimizer,
     device,
     freeze_params=None,
+    freeze_fun=None,
```

Comment:
This is all very specific to your work. It would be best to move this function to the continuous learning framework, and not change this function.